src/sentry/api/serializers/models/notification_setting2.py

```diff
@@ -0,0 +1,112 @@
+from collections import defaultdict
+from typing import Any, Iterable, Mapping, MutableMapping, Optional, Set, Union
+
+from sentry.api.serializers import Serializer
+from sentry.models.notificationsetting import NotificationSetting
+from sentry.models.team import Team
+from sentry.models.user import User
+from sentry.notifications.types import NotificationSettingTypes
+from sentry.services.hybrid_cloud.organization import RpcTeam
+from sentry.services.hybrid_cloud.user import RpcUser
+
+
+class NotificationSettingsSerializer(Serializer):
+    """
+    This Serializer fetches and serializes NotificationSettings for a list of
+    targets (users or teams.) Pass filters like `project=project` and
+    `type=NotificationSettingTypes.DEPLOY` to kwargs.
+    """
+
+    def get_attrs(
+        self,
+        item_list: Iterable[Union["Team", "User"]],
+        user: User,
+        **kwargs: Any,
+    ) -> MutableMapping[Union["Team", "User"], MutableMapping[str, Set[Any]]]:
+        """
+        This takes a list of recipients (which are either Users or Teams,
+        because both can have Notification Settings). The function
+        returns a mapping of targets to flat lists of object to be passed to the
+        `serialize` function.
+
+        :param item_list: Either a Set of User or Team objects whose
+            notification settings should be serialized.
+        :param user: The user who will be viewing the notification settings.
+        :param kwargs: Dict of optional filter options:
+            - type: NotificationSettingTypes enum value. e.g. WORKFLOW, DEPLOY.
+        """
+        type_option: Optional[NotificationSettingTypes] = kwargs.get("type")
+
+        team_map = {t.id: t for t in item_list if isinstance(t, (Team, RpcTeam))}
+        user_map = {u.id: u for u in item_list if isinstance(u, (User, RpcUser))}
+
+        notifications_settings = NotificationSetting.objects._filter(
+            type=type_option,
+            team_ids=list(team_map.keys()),
+            user_ids=list(user_map.keys()),
+        )
+
+        result: MutableMapping[Union["Team", "User"], MutableMapping[str, Set[Any]]] = defaultdict(
```

Comment:
```suggestion
        result: MutableMapping[Union[Team, User], MutableMapping[str, Set[Any]]] = defaultdict(
```

You have the types available in this scope, so you shouldn't need string types.